SPT-808: Add Java schema-artefacts package generation

chrisclayson opened the pull request:
## What

- Add new Gradle `schema-artefacts` module
- Alter `generate_java_types.sh` to also generate separate schema
  generation in to a different directoy
- Amend `generate_json_schemas.sh` so as not generate `index.md` if
  template not present in target directory
- Add `VocabSchemaLoader` class to retreive schema from resources and
  return as `InputStream`

## Why

Allow generation and publishing of a package to Maven repo such that schemas can be imported into projects and managed through Dependabot.

chrisclayson committed:
SPT-808: Add Java schema-artefacts package publishing

- Add new Gradle `schema-artefacts` module
- Alter `generate_java_types.sh` to also generate separate schema
  generation in to a different directoy
- Amend `generate_json_schemas.sh` so as not generate `index.md` if
  template not present in target directory
- Add `VocabSchemaLoader` class to retreive schema from resources and
  return as `InputStream`

chris-dco commented on code-generators/java-types/schema-artefacts/build.gradle:
```diff
@@ -0,0 +1,46 @@
+}
+
+repositories {
+    mavenCentral()
```
Should probably be set to:
```groovy
repositories {
    maven {
        url 'https://gds.jfrog.io/artifactory/di-allowed-repos'
    }
}
```

chris-dco commented on code-generators/java-types/schema-artefacts/src/main/java/uk/gov/di/model/schema/VocabSchemaLoader.java:
```diff
@@ -0,0 +1,14 @@
+
+import java.io.InputStream;
+
+public class VocabSchemaLoader {
```
May be useful to have some predefined constants such as, so that you don't need to "know" the filename which could change:
```java
public static final String IDENTITY_CHECK_CREDENTIAL_JWT_NAME = "IdentityCheckCredentialJWT.json";
public static final String INHERITED_IDENTITY_JWT_NAME = "InheritedIdentityJWT.json";
public static final String RISK_ASSESSMENT_CREDENTIAL_JWT_NAME = "RiskAssessmentCredentialJWT.json";
public static final String SECURITY_CHECK_CREDENTIAL_JWT_NAME = "SecurityCheckCredentialJWT.json";
```

chris-dco commented on code-generators/java-types/settings.gradle:
```diff
@@ -16,3 +16,5 @@ dependencyResolutionManagement {
         }
     }
 }
+include 'schema-artefacts'
```
Can this be moved to the top with the other `include`?

chris-dco commented on code-generators/java-types/schema-artefacts/src/main/java/uk/gov/di/model/schema/VocabSchemaLoader.java:
```diff
@@ -0,0 +1,14 @@
+    private VocabSchemaLoader() {
+    }
+
+    public static InputStream getSchema(String name) {
```
May also be worth adding a method `getSchemaAsString`

chrisclayson replied:
Good idea.

chrisclayson committed:
SPT-808: Add getSchemaAsString and other review comments

- Add a `getSchemaAsString()` method
- Add schema name constants
- Re-order `settings.gradle`
- Use artefactory repository
